cpp/tests/io/csv_test.cpp

```diff
@@ -2194,4 +2194,31 @@ TEST_F(CsvReaderTest, DtypesMapInvalid)
   EXPECT_THROW(cudf_io::read_csv(in_opts), cudf::logic_error);
 }
 
+TEST_F(CsvReaderTest, CsvDefaultOptionsWriteReadMatch)
+{
+  auto const filepath = temp_env->get_temp_dir() + "issue.csv";
+
+  // make up some kind of dataframe
+  auto int_column = column_wrapper<int32_t>{10, 20, 30};
+  auto str_column = column_wrapper<cudf::string_view>{"abc", "mno", "xyz"};
+  cudf::table_view input_table(std::vector<cudf::column_view>{int_column, str_column});
+
+  // write that dataframe to a csv using default options to some temporary file
+  cudf_io::csv_writer_options writer_options =
+    cudf_io::csv_writer_options::builder(cudf_io::sink_info{filepath}, input_table);
+  cudf_io::write_csv(writer_options);
+
+  // read the temp csv file using default options
+  cudf_io::csv_reader_options read_options =
+    cudf_io::csv_reader_options::builder(cudf_io::source_info{filepath})
+      .header(-1)
```

Comment:
Test should not include this line. We should test here that the header is present even when the writer did not receive metadata. Setting header to -1 means that we are reading a file without a header.